ui/widgets/datepicker.js

```diff
@@ -334,10 +341,16 @@ var widget = $.widget( "ui.datepicker", {
 	},
 
 	_getParsedValue: function() {
-		return this.calendarInstance._parse( this.element.val() );
+		return this._parse( this.element.val() );
 	},
 
 	_setOption: function( key, value ) {
+		if ( key === "max" || key === "min" ) {
+			if ( $.type( value ) === "string" ) {
```

Comment:
jQuery's type detection is mostly unneeded in modern browsers. For strings, it was never needed. `$.type()` was just an abstraction away from the `$.is*()` methods which were limited to types that mattered.